### PR TITLE
Add prop.numberOfMonths comparison into componentDidUpdate and run this.setCalendarMonthWeeks if prop has changed.

### PR DESCRIPTION
**Pull request by LMulvey**
This change addresses #1394 to ensure that `adjustDayPickerHeight` is always being called with fresh data from `this.calendarMonthWeeks`.

In our project, we're using a function to return a `screenClass` depending on the current value of `window.innerWidth`. We use `screenClass` to dynamically set the `numberOfMonths` prop. Before `window.innerWidth` is set, `screenClass` returns `sm`, which causes the _initial_ value of `numberOfMonths` to be set to `1` and then immediately changed to `2`.

```
<ScreenClassRender
   render={screenClass => (
      <DayPickerRangeController
        {...props}
        numberOfMonths={
          ['md', 'lg', 'xl'].includes(screenClass) ? 2 : 1
        }
        onDatesChange={this.onDatesChange}
        onFocusChange={this.onFocusChange}
        focusedInput={focusedInput}
        startDate={startDate}
        endDate={endDate}
      />
    ) }
/>
```

Since `this.setCalendarMonthWeeks` was called in `componentDidMount` and not again when the prop changes, the values in `this.calendarMonthWeeks` were no longer fresh.

**Before fix:**
adjustmonthheightissues

**After fix:**
issuesfixed

I totally get that this may be a complete edge case and as a first-time contributor, I'm very open to any changes/criticism/etc with this.

Thanks!

### CODE REVIEW

File: test/components/DayPicker_spec.jsx [hunk 1]
```diff
@@ -790,11 +791,14 @@ describe('DayPicker', () => {
...
     describe('#componentDidUpdate', () => {
       let updateStateAfterMonthTransitionSpy;
+      let setCalendarMonthWeeksSpy;
+
       beforeEach(() => {
         updateStateAfterMonthTransitionSpy = sinon.stub(
           DayPicker.prototype,
           'updateStateAfterMonthTransition',
         );
+        setCalendarMonthWeeksSpy = sinon.stub(PureDayPicker.prototype, 'setCalendarMonthWeeks');
```
Comment (ljharb) [inline test/components/DayPicker_spec.jsx]: If you unskip the containing describe, do your new tests pass?

Comment (LMulvey) [reply test/components/DayPicker_spec.jsx]: Ah! They don't 👎 In fact, none of the tests in that section are passing (I didn't even notice the skip). It looks like it's been skipped for about two years now.

I can take it on and go back and fix all those tests. Should I fix mine for this PR and make a separate PR for fixing the other tests? Or lump it all into this one?

Comment (ljharb) [reply test/components/DayPicker_spec.jsx]: I don't mind if your new test happens to be skipped; but it's important that the new test *would* be passing, so that when we unskip this section, we don't have to go fix it :-)

Comment (LMulvey) [reply test/components/DayPicker_spec.jsx]: The issue here is that _all_ the lifecycle tests are being skipped because  `componentDidUpdate` isn't being called without fully mounting the component. Full mount isn't working right now and is throwing: `Error: It looks like you called `mount()` without a global document being loaded.`

I'm happy to implement something like `jsdom-global` to allow mounting to work with the setup but I'm wondering if that should be a separate PR from this one? Or if I should take another approach to testing these methods? I could obviously stub it out but that seems intuitive to me as that wouldn't _really_ be mimicking production.

Comment (ljharb) [reply test/components/DayPicker_spec.jsx]: If you could enable these tests in another PR, we could merge that prior to this one - that’d be great!

Comment (LMulvey) [reply test/components/DayPicker_spec.jsx]: #1415 addresses this - @ljharb

**Comment by haojonat** [comment]
Hi, I have created a new pr here: https://github.com/airbnb/react-dates/pull/2095
